SPARQLWrapper/main.py

```diff
@@ -136,18 +135,27 @@ def main(test=None):
     sparql.setReturnFormat(args.format)
     results = sparql.query().convert()
 
-    if args.format == "json":
+    ct = type(results)
+    if ct is dict:
+        # "json"
         print(json.dumps(results, indent=4))
-    elif args.format in ("xml", "rdf+xml", "json-ld"):
-        print(results.toxml())  # type: ignore
-    elif args.format == "n3":
-        g = rdflib.Graph()
-        g.parse(data=results, format="n3")  # type: ignore
-        print(g.serialize(format="n3"))
-    elif args.format in ("csv", "tsv", "turtle"):
-        print(results.decode("utf-8"))  # type: ignore
+    elif ct is xml.dom.minidom.Document:
+        # "xml", "rdf+xml", "json-ld"
+        print(results.toxml())
+    elif ct is bytes:
+        if args.format == "n3":
+            g = rdflib.Graph()
+            g.parse(data=results, format="n3")
+            print(g.serialize(format="n3"))
```

Comment:
Why serialize as n3 when in case of ConjunctivegGraph you use default serialization?

Reply:
Would it not be better to just leave output as is if it is already n3?

Reply:
fixed. c1484e337f2bb510f7ae03c343d00d9a131a4503